System/Random.hs

```diff
@@ -1083,6 +1084,29 @@ uniformIntegerM (l, h) gen
         let v' = v * b + fromIntegral x
         v' `seq` f (mag * b) v'
 
+-- | A slightly modified version of what can be found in [Lemire's
+-- paper](https://arxiv.org/pdf/1805.10941.pdf), [O'Neill's
+-- blogpost](https://www.pcg-random.org/posts/bounded-rands.html) and
+-- more directly from [O'Neill's github
+-- repo](https://github.com/imneme/bounded-rands/blob/3d71f53c975b1e5b29f2f3b05a74e26dab9c3d84/bounded32.cpp#L234).
+-- The modification is we want the range to be [0,t] not [0,t) to be
+-- compatible with unsignedBitmaskWithRejectionRM.
+unbiasedWordMult32 :: MonadRandom g m => Word32 -> g -> m Word32
+unbiasedWordMult32 s g
+  | s == maxBound = uniformWord32 g
+  | otherwise = go
+  where
+    r :: Word32
+    r = s + 1
```

Comment:
Could this spell trouble if `s == maxBound`?

Reply:
Yes which is why there is a guard at https://github.com/idontgetoutmuch/random/pull/65/files/01503a7aaef22af2b6e353908f0cfab0e0cdda20#diff-4a522d098452ebfc9be2d6291588f5bcR1096

Reply:
My bad! :man_facepalming: